test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/Roundtrip/JsonReaderWriterInjectionTests.cs

```diff
@@ -37,6 +38,59 @@ public void UseDefaultJsonReader()
             RunReaderTest(null, "{\"@odata.context\":\"http://test/$metadata#People/$entity\",\"PersonId\":999,\"Name\":\"Jack\"}");
         }
 
+        [Fact]
+        public void UseComplexJsonReaderNestedSelect()
+        {
+            // $select =Address/SubAddress/SubRoad OR
+            //$select =Address($select =SubAddress($select=SubRoad))
+
+            var resDict =  RunComplexReaderTest(null, "{\"@odata.context\":\"http://test/$metadata#People(Address/SubAddress/SubRoad)/$entity\","+
+            "\"Address\":{" +
+                "\"SubAddress\":{\"SubRoad\":\"Redmond\"}}}");
+
+            Assert.Equal(resDict["DefaultNs.SubAddress"].ToList()[0].Value, "Redmond");
+        }
+        
+        [Fact]
+        public void UseComplexJsonReaderExpandAndSelect()
+        {
+            // $Expand =PersonCity($select =Zipcode),Address/City

```

Comment:
>E [](start = 16, length = 1)

make the comments more clearly:
1) $expand not $Expand
2) remove the whitespace after $select

Reply:
Resolved

Reply:
are you resolved them? 

---
In reply to: [356794474](https://github.com/OData/odata.net/pull/1607#discussion_r356794474) [](ancestors = 356794474)

Reply:
REsolved